File: BuilderPattern/Java/CarBuilder/Car.java
package BuilderPattern.CarBuilder;

import java.awt.Color;

// Business Object
public class Car {

    // member all private, all final to make businessObject immutable
    private final String brand;
    private final String model;
    private final int amountOfDoors;
    private final Color color;

    // Constructor (includes a big amount of parameters and gets messy)
    // set to protected, so that access to this Constructor is denied outside of this package
    protected Car(String brand, String model, int amountOfDoors, Color color) {
        this.brand = brand;
        this.model = model;
        this.amountOfDoors = amountOfDoors;
        this.color = color;
    }

    // Getter and no Setter for data-integrity //
    public String getBrand() { return brand; }
    public String getModel() { return model; }
    public int getAmountOfDoors() { return amountOfDoors; }
    public Color getColor() { return color; }
    // Getter and no Setter for data-integrity //

    @Override
    public String toString(){
        return String.format("Brand: %s\nModel: %s\nAmountOfDoors: %d\nColor: %s", brand, model, amountOfDoors, color.toString());
    }
}

File: BuilderPattern/CSharp/NSCarBuilder/Car.cs
using System.Drawing;

namespace BuilderPattern.NSCarBuilder {
    // Business Object
    internal class Car {

        // member all private
        private string brand;
        private string model;
        private int amountOfDoors;
        private Color color;

        // Constructor (includes a big amount of parameters and gets messy)
        internal Car(string brand, string model, int amountOfDoors, Color color) {
            this.brand = brand;
            this.model = model;
            this.amountOfDoors = amountOfDoors;
            this.color = color;
        }

        // Properties with no setters for data integrity //
        public string Brand { get { return brand; } }
        public string Model { get { return model; } }
        public int AmountOfDoors { get { return amountOfDoors; } }
        public Color Color { get { return color; } }
        // Properties with no setters for data integrity //

        public override string ToString() {
            return string.Format("Brand: {0}\nModel: {1}\nAmountOfDoors: {2}\nColor: {3}", Brand, Model, AmountOfDoors, Color.ToString());
        }
    }
}